furnace/pid1.py

```diff
@@ -48,11 +55,33 @@ def enable_zombie_reaping(self):
         # and get rid of zombies automatically
         signal.signal(signal.SIGCHLD, signal.SIG_IGN)
 
+    @classmethod
+    def create_mount_target(cls, source, destination):
+        if source.is_file():
+            if destination.is_symlink():
+                destination.unlink()
+            destination.parent.mkdir(parents=True, exist_ok=True)
+            destination.touch()
+        else:
+            destination.mkdir(parents=True, exist_ok=True)
+
+    def create_bind_mounts(self):
+        for source, relative_destination, read_only in self.bind_mounts:
+            destination = self.root_dir.joinpath(relative_destination)
+            self.create_mount_target(source, destination)
+            mount(source, destination, None, MS_BIND, None)
+            if read_only:
+                # "Read-only bind mounts" are actually an illusion, a special feature of the kernel,
```

Comment:
This is not True anymore starting from libmount 2.27, see: https://unix.stackexchange.com/questions/128336/why-doesnt-mount-respect-the-read-only-option-for-bind-mounts
I checked, in Ubuntu xenial the libmount1 package version is 2.27.1-6ubuntu3.6. I'm not sure how old systems Furnace wants to support, but this can be done in one step without remount.

Reply:
Unfortunately that change only affects the mount cli command, not the mount syscall, or libc's mount function (which we use here)